### DISCUSSION TITLE
My program can't find my templates after pyinstaller

### DISCUSSION THREAD

**Discussion by luizhirasawa** [pyinstaller/pyinstaller#8223] [Help]
The .exe is running and starting the server but it always gives the message 

Template loader postmortem
Django tried loading these templates, in this order:

Using engine django:
    * django.template.loaders.app_directories.Loader: C:\Users\kirit\AppData\Local\Temp\_MEI73282\django\contrib\admin\templates\todos\home_page.html (Source does not exist)
    * django.template.loaders.app_directories.Loader: C:\Users\kirit\AppData\Local\Temp\_MEI73282\django\contrib\auth\templates\todos\home_page.html (Source does not exist)

Traceback (most recent call last):
  File "django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "todos\views.py", line 158, in todo_home
    return render(request, "todos/home_page.html", context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "django\shortcuts.py", line 24, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "django\template\loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Exception Type: TemplateDoesNotExist at /
Exception Value: todos/home_page.html

here's the repository link: https://github.com/artluiz/Projeto-Alho-da-Serra

If someone could help me I be very grateful, the project deadline is tomorrow

I'm using: pyinstaller manage.spec

pyinstaller manage.spec

**Comment by bwoodsend** [comment]
That repo link is private.

My advise is just the generic missing data files advice:
* Debug in onedir mode so that you can actually see if your data files are in your application
* If they're not, `--add-data` them in
* If they are, make sure your code to locate them is relocatable (doesn't use unanchored relative paths) and then adjust the *dest* component of your `--add-data` flags so that the data files are placed where your Python code is looking for it

That said, freezing a server is pointless and django is full of hacks that make it miserable to use with PyInstaller. Just don't do it!

**Reply by luizhirasawa** [reply]
I'll try man, thanks. I know it's a dumb thing to do with django, but my boss asked for this, so here I am hehe.

IT WORKED, I can run it now. But I triyed it in another dir, but it says the secret key is missing, because it's in my .env file, how to add this file?

**Reply by bwoodsend** [reply]
Not sure how much help I can be here. My experience with django goes as far as trying it a few times, acknowledging that it is the worst server library I have ever used (which is quite an accomplishment given how terrible flask is), and avoiding any situation that may require me to do anything more than scowl at it in the future...

It looks like that env file is supposed to go next to your `settings.py` which, if I'm correct in guessing that the `settings.py` is baked into your executable where you can no longer see it, would mean that you can't put the env file where it needs to go without embedding that into the application too (which defeats the purpose of having the file). It looks like you can pass filenames to `environ.Env.read_env()` – can you pass that filename into your application via a new command line option or another environment variable? It might not be the most conventional way of doing it but it looks to me like the most conventional way is flat out wrong.
